shared-helpers/src/utilities/pdfs.ts

```diff
@@ -1,20 +1,71 @@
-import { ListingEvent, ListingEventType } from "@bloom-housing/backend-core/types"
-import { FileServiceInterface, FileServiceProvider } from "@bloom-housing/shared-services"
+import { AssetCreate, ListingEvent, ListingEventType } from "@bloom-housing/backend-core/types"
 
 export const cloudinaryPdfFromId = (publicId: string, cloudName: string) => {
   return `https://res.cloudinary.com/${cloudName}/image/upload/${publicId}.pdf`
 }
 
+/**
+ * This function returns a URL that can be presented directly to the client for
+ * a public-facing Asset (specifically a PDF).  This is purely for presentation
+ * and has no bearing on any server-side functionality.
+ *
+ * Due to the way that assets are currently stored, not all assets contain a full
+ * URL and so we must instead use additional information to generate a URL from
+ * context.  This function is designed to work within the bounds of current logic
+ * while still functioning as expected should this logic then be implemented on
+ * the backend instead.
+ *
+ * This function is nearly identical to `getImageUrlFromAsset()` and they could
+ * likely be combined with minimal effort.  They are kept separate to adhere to
+ * the paradigm already established.
+ *
+ * Note that `cloudinaryCloudName` is a temporary (and optional) param used to
+ * pass in the "cloud name" for the cloudinary instance to use.  While it would
+ * not be unreasonable to use a `process.env` value directly on the function call,
+ * this implements a reasonable default while providing an easy way to override
+ * the value as needed without overly complcating the function signature.
+ *
+ * @param asset               The asset to get the URL for
+ * @param cloudinaryCloudName The "cloud name" for assets in Cloudinary
+ * @returns                   The URL to use to access the asset or null
+ */
+export const getPdfUrlFromAsset = (
+  asset: AssetCreate,
+  cloudinaryCloudName = process.env.CLOUDINARY_CLOUD_NAME || ""
+): string | null => {
+  const fileId = asset.fileId
+
+  // if fileId is a url, just return it
+  // putting this first enables us to enforce formatting on the backend
+  if (fileId.match(/https?:\/\//)) {
+    return fileId
+  }
+
+  // handle the specific case where it's a pdf stored in cloudinary
+  if (asset.label == "cloudinaryPDF") {
```

Comment:
I don't think this is always the case where a PDF always has that label. For instance paper applications do not have that label https://github.com/bloom-housing/bloom/blob/main/sites/partners/src/components/listings/PaperListingForm/sections/ApplicationTypes.tsx#L94

Reply:
Thanks for pointing that out.  That particular condition was based on the code that was there before (see removed line 15 below in the same file).  One of the challenges this set of PRs is trying to address (and the primary goal of this specific PR) is that the logic for determining how to format a URL is scattered across the entire repo.  Subsequent PRs are going to ultimately fold all of this logic into the backend such that the initial condition on line 40 above should be the only branch that gets evaluated.  

I suppose the most accurate implementation would only have two cases - raw URL or cloudinary file -  based on the fact that cloudinary was the only supported backend to date.  Given that we won't be using cloudinary ourselves, however, it's expected that none of the local listings will have anything that should be converted to cloudinary URL format.  That covers all of the partner site and most of the public site, with the exception being the listings fetched from outside.  Those are the main ones we need to worry about, but unfortunately we can't be 100% sure of the format since it does depend on env vars on a remote system.

Do you think a fair compromise then would be "if isUrl(fileId) then return fileId, otherwise return cloudinary(fileId)"?

Reply:
Thanks for explaining where that came from. In that case I think it makes sense to keep the current logic since we won't be using Cloudinary for new changes, so we can leave as is and check for "cloudinaryPdf".